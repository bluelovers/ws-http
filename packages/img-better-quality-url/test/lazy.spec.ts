import { basename, extname } from 'path';
import { betterQualityURL } from '../index';
import fixtureCanged from './fixture/canged';

describe('suppertd', () => {


	describe('can be changed', () =>
	{

		fixtureCanged.forEach(input => {
			test(input, () =>
			{

				let actual = betterQualityURL(input);

				expect(actual).toMatchSnapshot({
					changed: true,
					not_suppertd: false,
				});

			});
		})

	})

	describe('maybe not changed', () =>
	{
		[
			'https://img1.mitemin.net/k0/78/84kqizdv2mupf8n135o55zfol2wb_xan_by_go_2f0x.jpg',
		].forEach(input => {
			test(input, () =>
			{

				let actual = betterQualityURL(input);

				expect(actual).toMatchSnapshot({
					not_suppertd: false,
				});

			});
		})

	})


});
