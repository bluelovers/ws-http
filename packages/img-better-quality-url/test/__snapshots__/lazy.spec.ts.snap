// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`suppertd can be changed https://12243.mitemin.net/userpageimage/viewimagebig/icode/i366014/ 1`] = `
{
  "changed": true,
  "not_suppertd": false,
  "pd": {
    "domain": "mitemin",
    "hostname": "12243.mitemin.net",
    "icann": {
      "domain": "mitemin",
      "subDomains": [
        "12243",
      ],
      "topLevelDomains": [
        "net",
      ],
    },
    "labels": [
      "12243",
      "mitemin",
      "net",
    ],
    "subDomains": [
      "12243",
    ],
    "topLevelDomains": [
      "net",
    ],
    "type": "LISTED",
  },
  "url": "https://12243.mitemin.net/userpageimage/viewimage/icode/i366014",
}
`;

exports[`suppertd can be changed https://img1.mitemin.net/cv/po/bcxka2bnheig0k9hc67bh6orl_177v_1kw_146_7bey.jpg.580.jpg 1`] = `
{
  "changed": true,
  "not_suppertd": false,
  "pd": {
    "domain": "mitemin",
    "hostname": "img1.mitemin.net",
    "icann": {
      "domain": "mitemin",
      "subDomains": [
        "img1",
      ],
      "topLevelDomains": [
        "net",
      ],
    },
    "labels": [
      "img1",
      "mitemin",
      "net",
    ],
    "subDomains": [
      "img1",
    ],
    "topLevelDomains": [
      "net",
    ],
    "type": "LISTED",
  },
  "url": "https://img1.mitemin.net/cv/po/bcxka2bnheig0k9hc67bh6orl_177v_1kw_146_7bey.jpg",
}
`;

exports[`suppertd can be changed https://masiro.me/images/encode/other-210619172125-yXNz.jpg?quality=80 1`] = `
{
  "changed": true,
  "not_suppertd": false,
  "pd": {
    "domain": "masiro",
    "hostname": "masiro.me",
    "icann": {
      "domain": "masiro",
      "subDomains": [],
      "topLevelDomains": [
        "me",
      ],
    },
    "labels": [
      "masiro",
      "me",
    ],
    "subDomains": [],
    "topLevelDomains": [
      "me",
    ],
    "type": "LISTED",
  },
  "url": "https://masiro.me/images/encode/other-210619172125-yXNz.jpg",
}
`;

exports[`suppertd can be changed https://www.novelstar.com.tw/cache/cutw_300h_450/821739d2e7af6219b020ffabf4ca49371530349123.jpg.png 1`] = `
{
  "changed": true,
  "not_suppertd": false,
  "pd": {
    "domain": "novelstar",
    "hostname": "www.novelstar.com.tw",
    "icann": {
      "domain": "novelstar",
      "subDomains": [
        "www",
      ],
      "topLevelDomains": [
        "com",
        "tw",
      ],
    },
    "labels": [
      "www",
      "novelstar",
      "com",
      "tw",
    ],
    "subDomains": [
      "www",
    ],
    "topLevelDomains": [
      "com",
      "tw",
    ],
    "type": "LISTED",
  },
  "url": "https://www.novelstar.com.tw/cache/cutw_460h_690/821739d2e7af6219b020ffabf4ca49371530349123.jpg.png",
}
`;

exports[`suppertd maybe not changed https://img1.mitemin.net/k0/78/84kqizdv2mupf8n135o55zfol2wb_xan_by_go_2f0x.jpg 1`] = `
{
  "changed": false,
  "not_suppertd": false,
  "pd": {
    "domain": "mitemin",
    "hostname": "img1.mitemin.net",
    "icann": {
      "domain": "mitemin",
      "subDomains": [
        "img1",
      ],
      "topLevelDomains": [
        "net",
      ],
    },
    "labels": [
      "img1",
      "mitemin",
      "net",
    ],
    "subDomains": [
      "img1",
    ],
    "topLevelDomains": [
      "net",
    ],
    "type": "LISTED",
  },
  "url": "https://img1.mitemin.net/k0/78/84kqizdv2mupf8n135o55zfol2wb_xan_by_go_2f0x.jpg",
}
`;
